Ports/SDL2/patches/0001-Add-SerenityOS-platform-support.patch

```diff
@@ -469,20 +470,29 @@ index 4ea7705e959b8a2d120b4d1c8599c93f450b7d9f..83b9bba4ff81ff7030f2f6fa2235c916
  /* Use SDL_OnVideoThread() sparingly, to avoid regressions in use cases that currently happen to work */
  extern SDL_bool SDL_OnVideoThread(void);
 diff --git a/src/video/SDL_video.c b/src/video/SDL_video.c
-index 21c516265f75b61ec11aeb50e5a8e6cad7a44dd0..2e765deecce851935434c17f662463607fc7fadf 100644
+index 8065a0b8cfc71e8d4dadd7d40fbe15303ce7e05d..7514625ed8c4eb0d63a6bed61781ba6410acfc5a 100644
 --- a/src/video/SDL_video.c
 +++ b/src/video/SDL_video.c
-@@ -128,6 +128,9 @@ static VideoBootStrap *bootstrap[] = {
-     &OS2DIVE_bootstrap,
-     &OS2VMAN_bootstrap,
+@@ -138,6 +138,9 @@ static VideoBootStrap *bootstrap[] = {
+ #if SDL_VIDEO_DRIVER_OFFSCREEN
+     &OFFSCREEN_bootstrap,
  #endif
 +#if SDL_VIDEO_DRIVER_SERENITY
 +    &SERENITYVIDEO_bootstrap,
 +#endif
  #if SDL_VIDEO_DRIVER_DUMMY
      &DUMMY_bootstrap,
  #if SDL_INPUT_LINUXEV
-@@ -4258,6 +4261,10 @@ SDL_WM_SetIcon(SDL_Surface * icon, Uint8 * mask)
+@@ -466,7 +469,7 @@ int SDL_VideoInit(const char *driver_name)
+     /* Select the proper video driver */
+     video = NULL;
+     if (driver_name == NULL) {
+-        driver_name = SDL_GetHint(SDL_HINT_VIDEODRIVER);
++        driver_name = "serenity";
```

Comment:
This is the only real change I had to make to get it working (upstream commit here: https://github.com/libsdl-org/SDL/commit/c1336b21). 

There's almost certainly a more correct way to do this, but since we're always going to be using the "serenity" driver, I thought this was fine.

Reply:
Fwiw, I think it's fine to ignore the hint for now and set "serenity" as a default. For other platforms it's possible to specify a priority list of video drivers, but since we always have to force "serenity" as a result we can just ignore the hint completely.

Reply:
The real issue here is that our `serenity` video driver is tested after `offscreen`, which is a headless video driver that always works. We simply need to reorder the list of video drivers so that we have a higher priority than that.

Reply:
@timschumi My latest push updates the PR to do bootstrapping of the serenity audio and video drivers before any others, as suggested on Discord. I've tested this with all the games mentioned above and all work as expected :^).